src/Sentry/Protocol/Transaction.cs

```diff
@@ -0,0 +1,329 @@
+
+        /// <summary>
+        /// Transaction name.
+        /// </summary>
+        public string Name { get; set; } = "unnamed";
+
+        /// <inheritdoc />
+        public SentryId SpanId
+        {
+            get => Contexts.Trace.SpanId;
+            private set => Contexts.Trace.SpanId = value;
+        }
+
+        /// <inheritdoc />
+        public SentryId TraceId
+        {
+            get => Contexts.Trace.TraceId;
+            private set => Contexts.Trace.TraceId = value;
+        }
+
+        /// <inheritdoc />
+        public DateTimeOffset StartTimestamp { get; private set; } = DateTimeOffset.UtcNow;
+
+        /// <inheritdoc />
+        public DateTimeOffset? EndTimestamp { get; private set; }
+
+        /// <inheritdoc />
+        public string Operation
+        {
+            get => Contexts.Trace.Operation;
+            internal set => Contexts.Trace.Operation = value;
+        }
+
+        /// <inheritdoc />
+        public string? Description { get; set; }
+
+        /// <inheritdoc />
+        public SpanStatus? Status
+        {
+            get => Contexts.Trace.Status;
+            private set => Contexts.Trace.Status = value;
+        }
+
+        /// <inheritdoc />
+        public bool IsSampled { get; set; }
+
+        /// <inheritdoc />
+        public SentryLevel? Level { get; set; }
+
+        private Request? _request;
+
+        /// <inheritdoc />
+        public Request Request
+        {
+            get => _request ??= new Request();
+            set => _request = value;
+        }
+
+        private Contexts? _contexts;
+
+        /// <inheritdoc />
+        public Contexts Contexts
+        {
+            get => _contexts ??= new Contexts();
+            set => _contexts = value;
+        }
+
+        private User? _user;
+
+        /// <inheritdoc />
+        public User User
+        {
+            get => _user ??= new User();
+            set => _user = value;
+        }
+
+        /// <inheritdoc />
+        public string? Environment { get; set; }
+
+        /// <inheritdoc />
+        public SdkVersion Sdk { get; internal set; } = new SdkVersion();
+
+        private IEnumerable<string>? _fingerprint;
+
+        /// <inheritdoc />
+        public IEnumerable<string> Fingerprint
+        {
+            get => _fingerprint ?? Enumerable.Empty<string>();
+            set => _fingerprint = value;
+        }
+
+        private List<Breadcrumb>? _breadcrumbs;
+
+        /// <inheritdoc />
+        public IEnumerable<Breadcrumb> Breadcrumbs => _breadcrumbs ??= new List<Breadcrumb>();
+
+        private Dictionary<string, object?>? _extra;
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<string, object?> Extra => _extra ??= new Dictionary<string, object?>();
+
+        private Dictionary<string, string>? _tags;
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<string, string> Tags => _tags ??= new Dictionary<string, string>();
```

Comment:
Not thread-safe.

Same as other discussions, Lazy<T> or assigning Concurrent.X from instantiation, or double check lock, or Interlocked.. lets hop on a call and think this through to choose the right approach?